dask_mongo/core.py

```diff
@@ -9,24 +9,44 @@
 from dask.bag import Bag
 from dask.base import tokenize
 from dask.graph_manipulation import checkpoint
-
 from ._version import __version__
+from functools import lru_cache
 
 appname = f"dask-mongo/{__version__}"
 
+_CACHE_SIZE = 10
+
+
+class CachedKwargs(dict):
```

Comment:
Can you explain how this deals with unhashable MongoClient args (eg event_listeners)?